Destiny.ScrimTracker.Logic/Repositories/GuardianRepository.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Destiny.ScrimTracker.Logic.Adapters;
+using Destiny.ScrimTracker.Logic.Models;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Query;
+
+namespace Destiny.ScrimTracker.Logic.Repositories
+{
+    public interface IGuardianRepository
+    {
+        string CreateGuardian(Guardian guardian);
+        IEnumerable<Guardian> GetAllGuardians();
+        Guardian GetGuardian(string guardianId);
+        Guardian UpdateGuardianStats(Guardian updatedGuardian);
+        string DeleteGuardian(string guardianId);
+    }
+    
+    public class GuardianRepository : IGuardianRepository
+    {
+        private readonly GuardianContext _guardianContext;
+
+        public GuardianRepository(GuardianContext guardianContext)
+        {
+            _guardianContext = guardianContext;
+        }
+
+        public string CreateGuardian(Guardian guardian)
+        {
+            using (var context = _guardianContext)
+            {
+                context.Add(guardian);
+                context.SaveChanges();
+            }
+
+            return guardian.Id;
+        }
+
+        public IEnumerable<Guardian> GetAllGuardians()
+        {
+            using (var context = _guardianContext)
+            {
+                var guardians =  context.Guardians.ToList();
```

Comment:
Where is this method `Guardians` defined on `GuardianContext`?

Reply:
It's not a method, it is a collection of objects. Entity Framework (our ORM) knows that when we are trying to access this collection, to read it in from the database first using LINQ (Language Integrated Query)